### PR TITLE
feat: support for github pull request

### PR DESCRIPTION
**Pull request by conwnet**

### CODE REVIEW

**Comment by conwnet** [comment]
demo: https://github1s-git-feature-github-pull-vscode-github1s.vercel.app/conwnet/github1s/pull/228

File: extensions/github1s/src/source-control/changes.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+		const pullFiles = await repository.getPullFiles(routerState.pullNumber);
+
+		return pullFiles.map((pullFile) => {
+			// the `previous_filename` field only exists in `RENAMED` file,
+			// fallback to `filename` otherwise
+			const previousFilePath = pullFile.previous_filename || pullFile.filename;
+			const currentFilePath = pullFile.filename;
+
+			return {
+				previousFileUri: vscode.Uri.joinPath(baseRootUri, previousFilePath),
```
Comment (xcv58) [inline extensions/github1s/src/source-control/changes.ts:49]: It fails to handle rename file:

https://github1s-git-feature-github-pull-vscode-github1s.vercel.app/xcv58/github1s/pull/16

image

Comment (xcv58) [reply extensions/github1s/src/source-control/changes.ts:49]: It might be caused by a bug on the GitHub side.
Let me reproduce.

Comment (xcv58) [reply extensions/github1s/src/source-control/changes.ts:49]: Confirmed it's a bug on the GitHub side https://github1s-git-feature-github-pull-vscode-github1s.vercel.app/xcv58/github1s/pull/17

**Review by xcv58 — APPROVED** [review]

**Review by Siddhant-K-code — APPROVED** [review]

**Pull request merged**